src/feature_detection/detectors.h

```diff
@@ -0,0 +1,78 @@
+#ifndef DETECTORS_H
+#define DETECTORS_H
+
+#include <keypoint_t/keypoint_t.h>
+#include <keypoints/keypoints.h>
+#include <math/math.h>
+#include <matrix_t/matrix_t.h>
+#include <types/types.h>
+#include <yape06/yape06.h>
+
+namespace jsfeat {
+
+namespace detectors {
+
+class Detectors : public Yape06, public Math {
+ public:
+  Detectors() {}
+  ~Detectors() {}
+
+  int detect_keypoints(Matrix_t* img, KeyPoints corners, int max_allowed) {
+    // detect features
+    auto kpc = detect_internal(img, &corners, 17);
+    auto count = kpc.count;
+    // sort by score and reduce the count if needed
+    if (count > max_allowed) {
+      // qsort_internal<KeyPoint_t, bool>(corners.kpoints, 0, count - 1, [](KeyPoint_t i, KeyPoint_t j){return (i.score < j.score);});
```

Comment:
I'm not sure of this, maybe it's better to use another small different approach. I'm looking to the OpenCV code in the Orb implementation and there is another possibility.

Reply:
retainBest is taken from OpenCV, but i need to figure out if this is correct.